File: amateur-football-mvp/fix_jsx.py
import sys

filepath = r'c:\Users\andyc\OneDrive\Documentos\GitHub\pelotify\amateur-football-mvp\src\app\canchas\page.tsx'

with open(filepath, 'r', encoding='utf-8') as f:
    lines = f.readlines()

# We want to find the lines for the button closing and add a </div> after it.
# Line 1418 in view_file is index 1417.
# 1416:                 {isSavingPrices ? 'Guardando...' : 'Guardar Todo'}
# 1417:              </button>
# 1418:           </div>

# We will look for this pattern:
target_pattern = "{isSavingPrices ? 'Guardando...' : 'Guardar Todo'}"

for i, line in enumerate(lines):
    if target_pattern in line:
        # Check if lines[i+1] is </button> and lines[i+2] is </div>
        if '</button>' in lines[i+1] and '</div>' in lines[i+2]:
            # This is our spot.
            # Insert </div> after lines[i+2]
            lines.insert(i+3, '        </div>\n')
            print(f"Found and inserted at line {i+4}")
            break

with open(filepath, 'w', encoding='utf-8') as f:
    f.writelines(lines)

File: amateur-football-mvp/fix_jsx.js
const fs = require('fs');
const path = require('path');

const filepath = 'c:\\Users\\andyc\\OneDrive\\Documentos\\GitHub\\pelotify\\amateur-football-mvp\\src\\app\\canchas\\page.tsx';

try {
    const content = fs.readFileSync(filepath, 'utf8');
    const lines = content.split(/\r?\n/);

    const targetPattern = "{isSavingPrices ? 'Guardando...' : 'Guardar Todo'}";

    let inserted = false;
    for (let i = 0; i < lines.length; i++) {
        if (lines[i].includes(targetPattern)) {
            // Check following lines for button and div closing
            if (lines[i+1]?.includes('</button>') && lines[i+2]?.trim() === '</div>') {
                // This is where Pricing Card ends. Add Left Column closing after.
                lines.splice(i+3, 0, '           </div>');
                inserted = true;
                console.log(`Successfully added closing div after line ${i+3}`);
                break;
            }
        }
    }

    if (inserted) {
        fs.writeFileSync(filepath, lines.join('\n'), 'utf8');
    } else {
        console.error("Pattern not found to fix JSX.");
        process.exit(1);
    }
} catch (err) {
    console.error("Error fixing JSX: ", err);
    process.exit(1);
}
